src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,47 +1,79 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  selectedTodoId: number,
+  selectedUserId: number,
+  onReset: () => void,
+  todos: Todo[],
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodoId,
+  selectedUserId,
+  onReset,
+  todos,
+}) => {
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    getUser(selectedUserId)
+      .then(userFromServer => {
+        setUser(userFromServer);
+      });
+  }, []);
+
+  const findTodo = todos.filter(({ id }) => selectedTodoId === id);
```

Comment:
Maybe better use the find method?)

Reply:
do you need selectedUserId on App if here you find selected todo (todo has userId)?